### PR TITLE
[issue-426] add Dataset class

### PR DESCRIPTION
**Pull request by armintaenzertng**
part of #426

### CODE REVIEW

File: src/spdx_tools/spdx3/model/dataset/dataset.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    data_collection_process: Optional[str] = None
+    intended_use: Optional[str] = None
+    dataset_size: Optional[int] = None
+    dataset_noise: Optional[str] = None
+    data_preprocessing_steps: Optional[str] = None
+    sensors: Optional[Dict[str, str]] = None
+    known_biases: Optional[str] = None
+    sensitive_personal_information: Optional[bool] = None
+    anonymization_method_used: Optional[str] = None
+    confidentiality_level: Optional[ConfidentialityLevelType] = None
```
Comment (meretp) [inline src/spdx_tools/spdx3/model/dataset/dataset.py:37]: These 3 properties don't have a maxCount, so I would expect lists here.

File: src/spdx_tools/spdx3/model/dataset/dataset.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+
+@dataclass_with_properties
+class Dataset:
+    data_collection_process: Optional[str] = None
+    intended_use: Optional[str] = None
+    dataset_size: Optional[int] = None
+    dataset_noise: Optional[str] = None
+    data_preprocessing_steps: Optional[str] = None
+    sensors: Optional[Dict[str, str]] = None
```
Comment (meretp) [inline src/spdx_tools/spdx3/model/dataset/dataset.py:33]: This property doesn't have a maxCount neither, so I would expect a list here.

File: src/spdx_tools/spdx3/model/dataset/dataset.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+from spdx_tools.common.typing.dataclass_with_properties import dataclass_with_properties
+from spdx_tools.common.typing.type_checks import check_types_and_set_values
+
+
+class ConfidentialityLevelType(Enum):
+    RED = auto()
+    AMBER = auto()
+    GREEN = auto()
+    CLEAR = auto()
```
Comment (meretp) [inline src/spdx_tools/spdx3/model/dataset/dataset.py:15]: This one is not listed in the markdown file, but probably an issue in the spdx-3-model?

File: src/spdx_tools/spdx3/model/dataset/dataset.py (hunk 2 above)
Comment (meretp) [reply src/spdx_tools/spdx3/model/dataset/dataset.py:33]: I have one question concerning this: the markdown file for `DictionaryEntry` states that the property `value` is optional, how would we handle this? I would think `List[Dict[str, Optional[str]]]` would be the correct type hint.

**Review by meretp — CHANGES_REQUESTED** [review]
Thanks! I have some comments/ remarks concerning the type hints, not sure if these are issues in the spec or in this PR.

File: src/spdx_tools/spdx3/model/dataset/dataset.py (hunk 1 above)
Comment (armintaenzertng) [reply src/spdx_tools/spdx3/model/dataset/dataset.py:37]: Good point for `anonymization_method_used`; for the other two I [opened a PR](https://github.com/spdx/spdx-3-model/pull/219) as I think they actually should have a `maxCount`.

File: src/spdx_tools/spdx3/model/dataset/dataset.py (hunk 3 above)
Comment (armintaenzertng) [reply src/spdx_tools/spdx3/model/dataset/dataset.py:15]: Yes, I opened an issue for this.

File: src/spdx_tools/spdx3/model/dataset/dataset.py (hunk 2 above)
Comment (meretp) [reply src/spdx_tools/spdx3/model/dataset/dataset.py]: The only important difference would be the need for unique keys. So I don't know if this makes any sense, but from the definition in the spdx-3-model we could have two sensors with the same key, don't we?

Comment (armintaenzertng) [reply src/spdx_tools/spdx3/model/dataset/dataset.py]: I believe this is still debated whether this is actually wanted in the spdx-3-model... If yes, then you are right, we have a problem

Comment (armintaenzertng) [reply src/spdx_tools/spdx3/model/dataset/dataset.py]: Removed the `Optional`

**Review by meretp — APPROVED** [review]
lgtm

**Pull request merged**